pkg/oc/bootstrap/docker/openshift/templateservicebroker.go

```diff
@@ -0,0 +1,63 @@
+package openshift
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+
+	kapierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	kapi "k8s.io/kubernetes/pkg/api"
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/oc/bootstrap/docker/errors"
+)
+
+const (
+	tsbNamespace        = "openshift-template-service-broker"
+	tsbTemplateName     = "template-service-broker"
+	tsbTemplateLocation = "test/testdata/templateservicebroker/templateservicebroker-template.yaml"
+)
+
+// InstallServiceCatalog checks whether the template service broker is installed and installs it if not already installed
+func (h *Helper) InstallTemplateServiceBroker(f *clientcmd.Factory) error {
+	osClient, kubeClient, err := f.Clients()
+	if err != nil {
+		return errors.NewError("cannot obtain API clients").WithCause(err).WithDetails(h.OriginLog())
+	}
+
+	// create the namespace if needed.  This is a reserved namespace, so you can't do it with the create project request
+	if _, err := kubeClient.Core().Namespaces().Create(&kapi.Namespace{ObjectMeta: metav1.ObjectMeta{Name: tsbNamespace}}); err != nil && !kapierrors.IsAlreadyExists(err) {
+		return errors.NewError("cannot create template service broker project").WithCause(err)
+	}
+
+	// create the template in the tsbNamespace to make it easy to instantiate
+	if err := ImportObjects(f, tsbNamespace, tsbTemplateLocation); err != nil {
+		return errors.NewError("cannot create template service broker template").WithCause(err)
+	}
+
+	// create the actual resources required
+	if err = instantiateTemplate(osClient, clientcmd.ResourceMapper(f), tsbNamespace, tsbTemplateName, tsbNamespace, nil, true); err != nil {
+		return errors.NewError("cannot instantiate logger accounts").WithCause(err)
+	}
+
+	// Wait for the apiserver endpoint to become available
+	err = wait.Poll(5*time.Second, 600*time.Second, func() (bool, error) {
```

Comment:
> If you're saying that the api becomes available before the service catalog actually considers itself ready (and that the SC isn't ready until significantly later) that's news to me, i've never seen an issue doing oc cluster up and then going straight to the web console.

Takes maybe 15 seconds.  The service catalog is running, but the service catalog hasn't filled in any status (good or bad) on the broker resource.